homeassistant/components/plugwise/config_flow.py

```diff
@@ -44,12 +62,47 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
 
+    def __init__(self):
+        """Initialize the Plugwise config flow."""
+        self.discovery_info = {}
+
+    async def _async_set_unique_id(self, unique_id):
+        """Helperfunction setting the config entry's unique ID."""
+        await self.async_set_unique_id(unique_id)
+        self._abort_if_unique_id_configured()
+
+    async def async_step_zeroconf(self, discovery_info: DiscoveryInfoType):
+        """Prepare configuration for a discovered Plugwise Smile."""
+        self.discovery_info = discovery_info
+        _properties = self.discovery_info.get("properties")
+
+        unique_id = self.discovery_info.get("hostname").split(".")[0]
+        await self._async_set_unique_id(unique_id)
+
+        _product = _properties.get("product", None)
+        _version = _properties.get("version", "n/a")
+        _name = f"{ZEROCONF_MAP.get(_product,_product)} v{_version}"
+
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        self.context["title_placeholders"] = {
+            CONF_HOST: discovery_info[CONF_HOST],
+            "name": _name,
+        }
+        return await self.async_step_user()
+
     async def async_step_user(self, user_input=None):
         """Handle the initial step."""
         errors = {}
 
+        for entry in self._async_current_entries():
+            if entry.unique_id == self.discovery_info.get("hostname").split(".")[0]:
+                self._abort_if_unique_id_configured()
```

Comment:
This is not necessary, since you already set it as unique ID in the zeroconf step.

Reply:
This bit of code has been added to make discovery and configuration work when there are 2 or more Smiles present in the network. Both @CoMPaTech and I have 2 Smiles in our network.

Reply:
I'll retest :)

Reply:
Testing shows that the above 3 lines of code can be removed.